File: src/main/java/me/kaotich00/fwauctionhouse/storage/StorageMethod.java
package me.kaotich00.fwauctionhouse.storage;

import me.kaotich00.fwauctionhouse.FwAuctionHouse;
import me.kaotich00.fwauctionhouse.objects.PendingSell;
import org.bukkit.entity.Player;
import org.bukkit.inventory.ItemStack;

import java.sql.Connection;
import java.sql.SQLException;
import java.util.List;

public interface StorageMethod {

    FwAuctionHouse getPlugin();

    void init();

    void shutdown();

    Connection getConnection() throws SQLException;

    void insertListing(Player seller, ItemStack itemStack, Float unitPrice);

    List<PendingSell> getPendingSells();

    void updateListingStatus(int listingId, int status);

    void deletePendingSell(int listingId);

}

File: src/main/java/me/kaotich00/fwauctionhouse/storage/StorageMethod.kt
package me.kaotich00.fwauctionhouse.storage

import me.kaotich00.fwauctionhouse.FwAuctionHouse
import me.kaotich00.fwauctionhouse.objects.PendingSell
import me.kaotich00.fwauctionhouse.objects.PendingToken
import org.bukkit.entity.Player
import org.bukkit.inventory.ItemStack
import java.sql.Connection

interface StorageMethod {

    val plugin: FwAuctionHouse

    fun init()

    fun shutdown()

    val connection: Connection

    fun insertListing(seller: Player, itemStack: ItemStack, unitPrice: Double)

    fun getPendingSells(): List<PendingSell>

    fun updateListingStatus(listingId: Int, status: Int)

    fun deletePendingSell(listingId: Int)

    fun getPendingTokens(): List<PendingToken>

    fun validateToken(sessionId: Int)
}